backpack/extensions/backprop_extension.py

```diff
@@ -54,10 +55,19 @@ def __init__(self, savefield, module_exts, fail_mode=FAIL_ERROR):
 
     @classmethod
     def add_module_extension(cls, module, extension):
+        """Adds an external module mapping.
+
+        Args:
+            module(Type[torch.nn.Module]): The module that is supposed to be extended
+            extension(backpack.extensions.module_extension.ModuleExtension):
+                The custom extension of that module.
+        """
         cls.__external_module_extensions[module] = extension
 
     def __get_module_extension(self, module):
         module_extension = self.__module_extensions.get(module.__class__)
+        if module_extension is None:
+            module_extension = self.__external_module_extensions.get(module.__class__)
```

Comment:
We need to double-check something here. My python knowledge is not solid enough to tell whether, if called from a child class, this method will write to `BackpropExtension.__external_module_extension`, or the child class' attribute.

```python
# writes to BatchGrad.__external_module_extensions or BackpropExtension.__external_module_extensions?
BatchGrad.add_module_extension(MyLinearImplementation1, BatchGradLinear) 
# writes to Grad.__external_module_extensions or BackpropExtension.__external_module_extensions?
# If the second happens, the first call will be overwritten and thus effectively eliminated
Grad.add_module_extension(MyLinearImplementation1, GradLinear)
```
Needs to be debugged before merging. I suspect it has to be refactored because I already experienced the problem for another project.

Reply:
A quick test: A has a class attribute number, and a classmethod set_number. B and C inherit from A.
```
print(A.number, B.number, C.number)  # 0,0,0
A.set_number(1)
print(A.number, B.number, C.number)  # 1,1,1
B.set_number(2)
print(A.number, B.number, C.number)  # 1,2,1
C.set_number(3)
print(A.number, B.number, C.number)  # 1,2,3
A.set_number(0)
print(A.number, B.number, C.number)  # 0,2,3
```
This result is surprising to me. My explanation is that initially the attributes of B and C point to the same attribute as A. But as soon as they are changed they point to a different object.
Thus, as far as our code is concerned, it is fine if the child class adds the module extension. Should we add an error if the method is called from the parent class BackpropExtension?

Reply:
Indeed this will cause a bug for someone in the future. I think the easiest, though not most elegant, solution is to make ``__external_module_extension`` independent among instances (i.e. assign it in the constructor) and make ``add_module_extensions`` a regular method of ``BackpropExtension``. What do you think?

Reply:
I would take the following precautions:
1. Against accidently initializing in BackpropExtension instead of an inherited class - check the type of class.
2. Against accidently overwriting an existing mapping - check if the mapping already exists.

I also realized, that the dictionary is added to the regular dictionary in  ``__init__``. Therefore, the check in ``__get_module_extension`` is not necessary.

If you think this solution is not sufficient, I will change the variable to a regular attribute.

Reply:
After discussion, removed ``__external_module_extension``.